File: shoring/blackjackEngine/src/test/java/com/panopset/tests/blackjackEngine/AbstractBlackjackTest.java
package com.panopset.tests.blackjackEngine;

import java.util.List;
import org.junit.jupiter.api.Assertions;
import com.panopset.blackjackEngine.BlackjackCard;
import com.panopset.blackjackEngine.BlackjackGameEngine;
import com.panopset.blackjackEngine.Card;
import com.panopset.blackjackEngine.CardDefinition;

public abstract class AbstractBlackjackTest implements PlayerScenarios  {

  public void verifyRecommendedActions(BlackjackGameEngine bge, String[] expectedActions,
      List<Card> stackedDeckForTesting) {
    new ScenarioVerifier().verifyRecommendedActions(bge, expectedActions, stackedDeckForTesting);
  }
  
  public void performDeviantActions(BlackjackGameEngine bge, String[] expectedActions,
      List<Card> stackedDeckForTesting) {
    new ScenarioVerifier().performDeviantActions(bge, expectedActions, stackedDeckForTesting);
  }

  @SafeVarargs
  public final void verifyRecommendedActionsFastDeal(BlackjackGameEngine bge, String[] expectedActions,
      List<Card>... stackedDeckForTesting) {
    new ScenarioVerifier().verifyRecommendedActionsFastDeal(bge, expectedActions, stackedDeckForTesting);
  }
  
  public void verifyPlayerCards(CardDefinition[] expected, List<BlackjackCard> cards) {
    if (expected == null || cards == null) {
      Assertions.fail();
    }
    Assertions.assertEquals(expected.length, cards.size());
    int i = 0;
    for (BlackjackCard blackjackCard : cards) {
      Assertions.assertEquals(expected[i++], blackjackCard.getCard().getCardDefinition());
    }
  }
}

File: code/shoring/blackjackEngine/src/test/kotlin/com/panopset/tests/blackjackEngine/AbstractBlackjackTest.kt
package com.panopset.tests.blackjackEngine

import com.panopset.blackjackEngine.BlackjackCard
import com.panopset.blackjackEngine.BlackjackGameEngine
import com.panopset.blackjackEngine.Card
import com.panopset.blackjackEngine.CardDefinition
import org.junit.jupiter.api.Assertions

fun verifyRecommendedActions(
    bge: BlackjackGameEngine, expectedActions: Array<String>,
    stackedDeckForTesting: List<Card>
) {
    ScenarioVerifier().verifyRecommendedActions(bge, expectedActions, stackedDeckForTesting)
}

fun performDeviantActions(
    bge: BlackjackGameEngine, expectedActions: Array<String>,
    stackedDeckForTesting: List<Card>
) {
    ScenarioVerifier().performDeviantActions(bge, expectedActions, stackedDeckForTesting)
}

@SafeVarargs
fun verifyRecommendedActionsFastDeal(
    bge: BlackjackGameEngine, expectedActions: Array<String>,
    vararg stackedDeckForTesting: List<Card>
) {
    ScenarioVerifier().verifyRecommendedActionsFastDeal(bge, expectedActions, *stackedDeckForTesting)
}

fun verifyPlayerCards(expected: Array<CardDefinition>, cards: List<BlackjackCard>) {
    Assertions.assertEquals(expected.size, cards.size)
    for ((i, blackjackCard) in cards.withIndex()) {
        Assertions.assertEquals(expected[i], blackjackCard.card.cardDefinition)
    }
}
